test/Npgsql.Tests/SchemaTests.cs

```diff
@@ -271,328 +284,231 @@ public async Task Precision_and_scale()
+        Assert.That(column["column_name"], Is.EqualTo("id"));
+        Assert.That(column["constraint_type"], Is.EqualTo("PRIMARY KEY"));
     }
 
     [Test]
-    [NonParallelizable]
     public async Task Primary_key_composite()
     {
-        using var conn = OpenConnection();
-        try
-        {
-            conn.ExecuteNonQuery(@"
-DROP TABLE IF EXISTS data;
-CREATE TABLE data (id1 INT, id2 INT, f1 INT);
-ALTER TABLE data ADD PRIMARY KEY (id1, id2);");
-            string[] restrictions = { null!, null!, "data" };
-            var dataTable = await GetSchema(conn, "CONSTRAINTCOLUMNS", restrictions);
-            var columns = dataTable.Rows.Cast<DataRow>()
-                .OrderBy(r => r["ordinal_number"]).ToList();
-
-            Assert.That(columns.All(r => r["table_schema"].Equals("public")));
-            Assert.That(columns.All(r => r["table_name"].Equals("data")));
-            Assert.That(columns.All(r => r["constraint_type"].Equals("PRIMARY KEY")));
-
-            Assert.That(columns[0]["column_name"], Is.EqualTo("id1"));
-            Assert.That(columns[1]["column_name"], Is.EqualTo("id2"));
-        }
-        finally
-        {
-            conn.ExecuteNonQuery("DROP TABLE IF EXISTS data");
-        }
+        await using var conn = await OpenConnectionAsync();
+        await using var _ = await CreateTempTable(conn, "id1 INT, id2 INT, f1 INT, PRIMARY KEY (id1, id2)", out var table);
+
+        var dataTable = await GetSchema(conn, "CONSTRAINTCOLUMNS", new[] { null, null, table });
+        var columns = dataTable.Rows.Cast<DataRow>().OrderBy(r => r["ordinal_number"]).ToList();
+
+        Assert.That(columns.All(r => r["table_schema"].Equals("public")));
+        Assert.That(columns.All(r => r["table_name"].Equals(table)));
+        Assert.That(columns.All(r => r["constraint_type"].Equals("PRIMARY KEY")));
+
+        Assert.That(columns[0]["column_name"], Is.EqualTo("id1"));
+        Assert.That(columns[1]["column_name"], Is.EqualTo("id2"));
     }
 
     [Test]
     public async Task Unique_constraint()
     {
-        using var conn = OpenConnection();
-        try
-        {
-            conn.ExecuteNonQuery(@"
-DROP TABLE IF EXISTS data;
-CREATE TABLE data (f1 INT, f2 INT);
-ALTER TABLE data ADD UNIQUE (f1, f2);");
-            string[] restrictions = { null!, null!, "data" };
-            var dataTable = await GetSchema(conn, "CONSTRAINTCOLUMNS", restrictions);
-            var rows = dataTable.Rows.Cast<DataRow>().ToList();
-        }
-        finally
-        {
-            conn.ExecuteNonQuery("DROP TABLE IF EXISTS data");
-        }
+        await using var conn = await OpenConnectionAsync();
+        await using var _ = await CreateTempTable(conn, "f1 INT, f2 INT, UNIQuE (f1, f2)", out var table);
+
+        var dataTable = await GetSchema(conn, "CONSTRAINTCOLUMNS", new[] { null, null, table });
+        var rows = dataTable.Rows.Cast<DataRow>().ToList();
```

Comment:
What exactly are we testing here?

Reply:
Not much (these tests are pretty old) :)

Adding some meaningful asserts.